### PR TITLE
Personal info

### PR DESCRIPTION
**Pull request by mctran-phi**
### What this PR does (required):
- Allow users to save their personal information to their account.

### Screenshots / Videos (required):
- Screen Shot 2021-07-21 at 10 34 58 PM

### Any information needed to test this feature (required):
- Go to the settings route and click on the personal information tab.

### CODE REVIEW

File: client/src/components/PersonalInfo/useStyles.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { makeStyles } from '@material-ui/core';
+
+const useStyles = makeStyles(() => ({
+  root: {
+    width: '70vw',
```
Comment (rajivtitus) [inline client/src/components/PersonalInfo/useStyles.ts]: Maybe add a maxWidth here too?

File: server/controllers/personalInfo.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+const PersonalInfo = require('../models/PersonalInfo');
+const asyncHandler = require("express-async-handler");
+
+exports.createPersonalInfo = asyncHandler(async (req, res) => {
+  try {
+    const info = PersonalInfo.create({
+      userId: req.user.id,
+      firstName: req.body.firstName,
```
Comment (rajivtitus) [inline server/controllers/personalInfo.js]: As almost all the fields names are similar here, you spread the `req.body` to make it look a lot cleaner.

File: server/models/PersonalInfo.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+const { Schema, model } = require('mongoose');
+
+const PersonalInfoSchema = new Schema({
```
Comment (rajivtitus) [inline server/models/PersonalInfo.js:3]: 👍🏽

File: server/controllers/personalInfo.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+const PersonalInfo = require('../models/PersonalInfo');
+const asyncHandler = require("express-async-handler");
+
+exports.createPersonalInfo = asyncHandler(async (req, res) => {
```
Comment (rajivtitus) [inline server/controllers/personalInfo.js]: Looks good for a start. However, under the current scenario, won't be creating a new document each time a user submits his profile info?

**Review by rajivtitus — COMMENTED** [review]
Good thinking by creating a separate model for personal info.
I've made some comments about implementation, please have a look. Thanks!

**Comment by mctran-phi** [comment]
I think I've fixed all the issues. Please take a look!

File: server/controllers/personalInfo.js [hunk 5]
```diff
@@ -3,17 +3,30 @@ const asyncHandler = require("express-async-handler");
...
-      birthday: req.body.birthday,
-      gender: req.body.gender,
-      about: req.body.about
+    const hasSubmitted = await PersonalInfo.findOne({
+      user: req.user.id
     });
+    if (hasSubmitted) {
+      hasSubmitted = {...req.body};
+      await hasSubmitted.save();
+      res.status(201).json(hasSubmitted);
```
Comment (rajivtitus) [inline server/controllers/personalInfo.js]: Please use an OK status code, 201 is for resource creation

File: server/controllers/personalInfo.js [hunk 6]
```diff
@@ -3,17 +3,30 @@ const asyncHandler = require("express-async-handler");
...
+  }
+});
+
+exports.getPersonalInfo = asyncHandler(async (req, res) => {
+  try {
+    const info = await PersonalInfo.find({
+      userId: req.user.id
+    });
+
     res.status(201).json(info);
```
Comment (rajivtitus) [inline server/controllers/personalInfo.js]: Same as above!

File: server/controllers/personalInfo.js [hunk 7]
```diff
@@ -3,17 +3,30 @@ const asyncHandler = require("express-async-handler");
...
-      userId: req.user.id,
-      firstName: req.body.firstName,
-      middleInit: req.body.middleInit,
-      lastName: req.body.lastName,
-      email: req.body.email,
-      phone: req.body.phone,
-      birthday: req.body.birthday,
-      gender: req.body.gender,
-      about: req.body.about
+    const hasSubmitted = await PersonalInfo.findOne({
```
Comment (rajivtitus) [inline server/controllers/personalInfo.js]: Maybe you could use fineOneAndUpdate here?

File: server/controllers/personalInfo.js [hunk 8]
```diff
@@ -3,17 +3,30 @@ const asyncHandler = require("express-async-handler");
...
+      user: req.user.id
     });
+    if (hasSubmitted) {
+      hasSubmitted = {...req.body};
```
Comment (rajivtitus) [inline server/controllers/personalInfo.js]: Won't spreading the `req.body` into `hasSubmitted` replace the entire object? (The document will no longer contain the userId in this case)

File: client/src/components/PersonalInfo/PersonalInfo.tsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+      middleInit: middleInit,
+      lastName: lastName,
+      email: email,
+      phone: phone,
+      dateOfBirth: dateOfBirth,
+      gender: gender,
+      about: about,
+    }
+    postPersonalInfo(info)
+      .then(res => console.log(res));
```
Comment (rajivtitus) [inline client/src/components/PersonalInfo/PersonalInfo.tsx]: Please remove

**Review by rajivtitus — COMMENTED** [review]
Thanks for making requested changes.

I've made 2-3 comments, please have a look. Thank you :)

**Comment by mctran-phi** [comment]
I've made the changes, let me know if there's anything else!

**Review by rajivtitus — APPROVED** [review]
LGTM! Thanks a lot for making requested changes :)

**Pull request closed without merging**